packages/vx-legend/src/types/index.ts

```diff
@@ -0,0 +1,71 @@
+// eslint doesn't know about @types/d3-scale
+// eslint-disable-next-line import/no-extraneous-dependencies
+import * as d3Scale from 'd3-scale';
+
+export type BaseInput = string | number | Date;
+export type BaseOutput = string | number | Date;
+
+export type ScaleLinear<Input, Output> = d3Scale.ScaleLinear<Input, Output>;
+export type ScaleOrdinal<Input, Output> = d3Scale.ScaleOrdinal<Input, Output>;
+export type ScaleBand<Input> = d3Scale.ScaleBand<Input>;
+export type ScaleThreshold<Input extends BaseInput, Output> = d3Scale.ScaleThreshold<Input, Output>;
+export type ScaleQuantile<Output> = d3Scale.ScaleQuantile<Output>;
+
+// @TODO BaseInput only needed for `ScaleThreshold`
```

Comment:
I'm also not sure how to get around this cleanly. only `ScaleThreshold` has restrictions on `Input`, but that vastly restricts uses of the others. @kristw do you have any advance `TS` ideas?

Reply:
might be able to use conditional types here

```ts
type ScaleType<Input, Output> = Input extends BaseInput
  ? ScaleThreshold<Input, Output> | NonThresholdType<Input, Output>
  : NonThresholdType<Input, Output>;
```

Reply:
I spent a lot of time on this and couldn't get the conditional typing to work 😱 I am TS-ignoring it for now and will sync with @kristw  and others when they're back from holiday.